index.js

```diff
@@ -150,15 +214,42 @@ const Web3 = require('web3');
       return amount;
     }
 
-    function createProposal(recipient, amount, url/*, metaData*/) {
-      return new Promise((resolve, reject) => {
-        // TODO write metaData to IPFS
-        robot.logger.debug(`Creating proposal to issue ${amount}₭S to ${recipient} for ${url}...`);
+    function createContributionDocument(contributor, url, description, details) {
+      let contribution = {
+        "@context": "https://schema.kosmos.org",
+        "@type": "Contribution",
+        contributor: {
+          ipfs: contributor.ipfsHash
+        },
+        kind: 'dev',
+        url: url,
+        description: description,
+        details: details
+      };
+
+      if (! tv4.validate(contribution, schemas["contribution"])) {
+        console.log('[kredits] invalid contribution data: ', util.inspect(contribution));
+        return Promise.reject('invalid contribution data');
+      }
 
+      return ipfs.add(new ipfs.Buffer(JSON.stringify(contribution)))
+        .then(res => { return res[0].hash; })
+        .catch(err => console.log(err));
+    }
+
+    function createProposal(recipient, amount, url, description, details) {
+      robot.logger.debug(`Creating proposal to issue ${amount}₭S to ${recipient} for ${url}...`);
+
+      return new Promise((resolve, reject) => {
+        // Get contributor details for GitHub user
         getContributorByGithubUser(recipient).then(c => {
-          kredits.addProposal(c.address, amount, url, '', (e/* , d */) => {
-            if (e) { reject(); return; }
-            messageRoom(`Let's give ${recipient} some kredits for ${url}! We just need two votes: https://kredits.kosmos.org`);
+          // Create document containing contribution data on IPFS
+          createContributionDocument(c, url, description, details).then(ipfsHash => {
+            // Create proposal on ethereum blockchain
+            kredits.addProposal(c.address, amount, url, ipfsHash, (e/* , d */) => {
+              if (e) { reject(e); return; }
+              messageRoom(`Let's give ${recipient} some kredits for ${url}: https://kredits.kosmos.org`);
```

Comment:
maybe we could move the messageRoom call outside of the createProposal call. Thus createProposal only does create the IPFS document and the proposal and resolve or reject and could be used somewhere else.